akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpHeader.scala

```diff
@@ -95,6 +95,10 @@ object HttpHeader {
       }
     } else ParsingResult.Error(ErrorInfo(s"Illegal HTTP header name", name))
 
+  def containsLineBreak(s: String): Boolean = {
+    s.contains("\n") || s.contains("\r") || s.contains("\r\n")
```

Comment:
Mmh, would be good to avoid traversing over the data multiple times. The latter check for `\r\n` should be redundant anyways.

You should be able to use `CharPredicate` for that. Maybe like that:

```scala
private val CRLF = CharPredicate('\r', '\n')

def validateHeaderValue(value: String): Boolean =
  // quote relevant RFC section here
  CRLF.matchesAny(s)
```